src/ensembl/src/content/app/browser/track-panel/track-panel-bar/TrackPanelBar.tsx

```diff
@@ -9,29 +25,41 @@ import chevronRightIcon from 'static/img/shared/chevron-right.svg';
 
 import styles from './TrackPanelBar.scss';
 
-type TrackPanelBarProps = {
-  activeGenomeId: string;
-  closeDrawer: () => void;
-  closeTrackPanelModal: () => void;
+type StateProps = {
+  activeGenomeId: string | null;
   isDrawerOpened: boolean;
   isTrackPanelModalOpened: boolean;
   isTrackPanelOpened: boolean;
   launchbarExpanded: boolean;
+  trackPanelModalView: string;
+};
+
+type DispatchProps = {
+  closeDrawer: () => void;
+  closeTrackPanelModal: () => void;
   openTrackPanelModal: (trackPanelModalView: string) => void;
   toggleTrackPanel: (isTrackPanelOpened?: boolean) => void;
-  trackPanelModalView: string;
 };
 
+type OwnProps = {};
```

Comment:
Why do we need this?

Reply:
It's an unofficial convention that we have been using to define `OwnProps` in connected components even if it doesn't have any properties. I wonder if this needs to change and also if we need to have an agreed convention for this. Any thoughts @azangru?

Reply:
Personally, I just throw everything into a single `Props` type; so if there is an unofficial convention, I have been violating it for quite a while now 😄 

I am mildly in favour of just a single Props type, because I can't see any real value behind splitting the Props type into the props that come from mapStateToProps, props that come from mapDispatchFromProps, and props that come from parent component.

Reply:
> Personally, I just throw everything into a single Props type; so if there is an unofficial convention, I have been violating it for quite a while now 😄
You probably have 😆 

The main reason why I chose to separate the prop types is so that it is easier to decipher where the individual props are coming from by just looking at the types.

On the other hand, I admit that letting TypeScript infer the type for `mapStateToProps` and `mapDispatchToProps` is better than writing it by hand.

Reply:
>  I admit that letting TypeScript infer the type for mapStateToProps and mapDispatchToProps is better than writing it by hand

Erm. You'll still have to write them by hand. Unless you are using the `useSelector` hook (😐), when it will probably infer the type of the thing returned by the selector.

Reply:
> On the other hand, I admit that letting TypeScript infer the type for mapStateToProps and mapDispatchToProps is better than writing it by hand.
By this, what I meant was writing these types separately for the constants that we use at the bottom of the file. For example:

`const mapDispatchToProps: DispatchProps = { updateTrackState };`

Reply:
Oh yes, we won't need to declare the type of the `mapDispatchToProps` object; `react-redux`'s types should do that for us.

Reply:
I have got rid of these different types for track panel related components.